src/cider/tests/utils/DuckDbQueryRunner.cpp

```diff
@@ -248,6 +250,38 @@ ::duckdb::Value duckDbValueAtVarcharBatch(const VarcharBatch* batch, int64_t off
     }                                                                        \
   }
 
+#define GEN_DUCK_DB_VALUE_FROM_ARROW_ARRAY_AND_SCHEMA_FUNC                               \
```

Comment:
Seems we don't use this convertor anymore. You can take a look at DuckDbArrowAdaptor.

Reply:
I just took a look at the code of DuckDbArrowAdaptor

> Besides the `DuckDbArrowAdaptor `part, the `ArrowChecker` is also in an initial state. We need to port the existing check logic. Do you plan to do it?

yes, only limit check support, I have no plans to do this.

Reply:
I just took a look at the code of DuckDbArrowAdaptor, seems it is useless, why need such class as duckdb already provide method to convert duckdb result to array schema? anyway, GEN_DUCK_DB_VALUE_FROM_ARROW_ARRAY_AND_SCHEMA_FUNC is used for insert ArrowArray data to duckdb.

Reply:
The adaptor is a workaround of an error to fetch duckdb result to Arrow, refer [PR](https://github.com/intel/BDTK/pull/77/files).  For input data of DuckDbRunner,  there is a method called DuckDbQueryRunner::createTableAndInsertArrowData to avoid current `GEN_DUCK_DB_VALUE_FROM_ARROW_ARRAY_AND_SCHEMA_FUNC`.

Reply:
> The adaptor is a workaround of an error to fetch duckdb result to Arrow, refer [PR](https://github.com/intel/BDTK/pull/77/files).
what's error? seems we don't set timezone before run duckdb sql

> For input data of DuckDbRunner, there is a method called DuckDbQueryRunner::createTableAndInsertArrowData to avoid current `GEN_DUCK_DB_VALUE_FROM_ARROW_ARRAY_AND_SCHEMA_FUNC`.

createTableAndInsertArrowData use CiderBatch as input, I just make a small change based on that function to accept ArrowArray as input

Reply:
> > The adaptor is a workaround of an error to fetch duckdb result to Arrow, refer [PR](https://github.com/intel/BDTK/pull/77/files).
> 
> what's error? seems we don't set timezone before run duckdb sql
> 
> > For input data of DuckDbRunner, there is a method called DuckDbQueryRunner::createTableAndInsertArrowData to avoid current `GEN_DUCK_DB_VALUE_FROM_ARROW_ARRAY_AND_SCHEMA_FUNC`.
> 
> createTableAndInsertArrowData use CiderBatch as input, I just make a small change based on that function to accept ArrowArray as input

The error is caused by duckdb version and as I looked at the `fetchDataToArrow` API, we probably can remove the adaptor later.